### PR TITLE
Qute: TemplateInstance - add createUni(), remove deprecated publisher()

### LINKED ISSUE #11026: Qute: document Mutiny usage in API
`io.quarkus.qute.TemplateInstance.createMulti()`

### PR DESCRIPTION
**Pull request by mkouba**
- also mention Mutiny in the docs
- resolves #11026

### CODE REVIEW

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -862,6 +863,50 @@ We can include the tag like this:
...
+      // consume the output...
+   } else {
+      // process failure...
+   }
+};
+----
+<1> Register a callback that is executed once the rendering is finished. 
+
+There are also two methods that return https://smallrye.io/smallrye-mutiny/[Mutiny] types.
+`TemplateInstance.createUni()` returns a new `Uni<String>` object.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/qute-reference.adoc:887]: What about `asUni`? and `asMulti`?

File: independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java [hunk 2]
```diff
@@ -79,6 +80,17 @@ public String render() {
                     }));
         }
 
+        @Override
+        public Uni<String> createUni() {
+            return Uni.createFrom().emitter(emitter -> renderAsync().whenComplete((r, f) -> {
```
Comment (cescoffier) [inline independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java:85]: When not just `Uni.createFrom().completionStage(() -> renderAsync())` ?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/qute-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/qute-reference.adoc:887]: Hm, I don't know. I usually only use `asX()` methods to easily cast a more generic type to a subtype, e.g. something like `org.jboss.jandex.AnnotationTarget.asMethod()`.

Comment (cescoffier) [reply docs/src/main/asciidoc/qute-reference.adoc:887]: as you have `renderAsync`, maybe `renderAsUni` and `renderAsMulti`?

File: independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java (hunk 2 above)
Comment (mkouba) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateImpl.java:85]: Ah, I missed the `UniCreate.completionStage(Supplier<? extends CompletionStage<? extends T>>)` variant. All the reactive APIs are so rich ;-).

File: docs/src/main/asciidoc/qute-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/qute-reference.adoc:887]: The thing is that `createUni()` and `createMulti()` don't trigger rendering. It's the subscription that does trigger it.

Comment (cescoffier) [reply docs/src/main/asciidoc/qute-reference.adoc:887]: Right! 

So, let's keep `createUni` and `createMulti`.

**Review by cescoffier — APPROVED** [review]

**Comment by mkouba** [comment]
Let's try to pass the **Maven Tests - JDK 11 Windows** job one more time. If it's cancelled again (timeout?), I'm going to merge this PR anyway.

**Pull request merged**